sending-a-message-to-a-service/README.md

```diff
@@ -0,0 +1,325 @@
+```
+
+- The call to the backend is done using a remote method call expression (using `->`), which distinguishes network calls from normal method calls. [Client data binding](https://ballerina.io/learn/by-example/http-client-data-binding/) is used to directly try and bind the JSON response on success to the expected array of records.
+
+```ballerina
+Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+```
+
+- Use the `is` check to decide the response based on the response to the client call. If the client call was successful and the respond payload was an array of `Doctor`s (as expected), then, directly return the array from the resource. If the request failed, send a "NotFound" response if the client call failed with a 4xx status code or return an "InternalServerError" response for other failures.
+
+```ballerina
+log:printInfo("Retrieving information", specialization = category);
+
+Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+if resp is Doctor[] {
+    return resp;
+}
+
+log:printError("Retrieving doctor information failed", resp);
+if resp is http:ClientRequestError {
+    return <http:NotFound> {body: string `category not found: ${category}`};
+}
+
+return <http:InternalServerError> {body: resp.message()};
+```
+
+#### Complete source
+
+You have successfully developed the required service.
+
+```ballerina
+import ballerina/http;
+import ballerina/log;
+
+type Doctor record {|
+    string name;
+    string hospital;
+    string category;
+    string availability;
+    decimal fee;
+|};
+
+configurable int port = 8290;
+configurable string healthcareBackend = "http://localhost:9090/healthcare";
+
+final http:Client queryDoctorEP = check initializeHttpClient();
+
+function initializeHttpClient() returns http:Client|error => new (healthcareBackend);
+
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() 
+            returns Doctor[]|http:NotFound|http:InternalServerError {
+        log:printInfo("Retrieving information", specialization = category);
+        
+        Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+        if resp is Doctor[] {
+            return resp;
+        }
+        
+        log:printError("Retrieving doctor information failed", resp);
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: string `category not found: ${category}`};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+#### Diagram
+
+The [sequence diagram view](https://wso2.com/ballerina/vscode/docs/implement-the-code/sequence-diagram-view/) for the implemented resource method is the following.
+
+![sequence diagram](./resources/tutorial_sending_a_message_to_a_service.png)
+
+### Step 3: Build and run the service
+
+You can run this service by navigating to the project root and using the `bal run` command.
+
+```bash
+sending-a-message-to-a-service$ bal run
```

Comment:
Do we need to show this project-name in the terminal? we have already mentioned in L185, right?

Reply:
I think it'll be clearer in case someone decides to skip the description and just look at the commands.

Reply:
Ack. +1